lib/packet.ml

```diff
@@ -14,28 +14,24 @@ type content_type =
   | ALERT
   | HANDSHAKE
   | APPLICATION_DATA
-  | HEARTBEAT
 
 let content_type_to_int = function
   | CHANGE_CIPHER_SPEC -> 20
   | ALERT -> 21
   | HANDSHAKE -> 22
   | APPLICATION_DATA -> 23
-  | HEARTBEAT -> 24
 and int_to_content_type = function
   | 20 -> Some CHANGE_CIPHER_SPEC
   | 21 -> Some ALERT
   | 22 -> Some HANDSHAKE
   | 23 -> Some APPLICATION_DATA
-  | 24 -> Some HEARTBEAT
   | _ -> None
```

Comment:
Now this is responsible for erroring out on `HEARTBEAT`?

Reply:
Yes, exactly. The `int_to_content_type` is used by `Reader.parse_record`, which errors out with `UnknownContent _` - which leads (in `Engine.separate_records` to `` Error (`Fatal (`UnknownContentType _)) ``.

There is still an open question about error semantics -- which cases are fatal, which are not. For the heartbeat, this requires a negotiation in the client/server hello (which we do not support) -- thus anyone sending us a heartbeat packet violates the protocol.